src/geoserver-create-imagemosaic-datastore/index.js

```diff
@@ -110,30 +117,29 @@ const geoserverCreateImageMosaicDatastore = async (workerJob, inputs) => {
     await execShellCommand('zip -j ' + zipPath + ' ' + fileToZip.join(' '));
 
 
-    // workaround to create a mosaic data store programmatically
-    // 1. we create the directory for the mosaic store
-    // 2. we add a dummy image to it
-    // 3. REST call to create image mosaic store
-    // 4. REST call to remove dummy image from GeoServer and database
-    // 5. remove dummy image from filesystem
-    const dummyImageName = '20220902T1352.tif';
-    const mosaicDir = path.join(geoserverDataDir, 'data', ws, covStore);
     log('Creating datastore directory');
+    const mosaicDir = path.join(geoserverDataDir, 'data', ws, covStore);
     await fsPromises.mkdir(mosaicDir, { recursive: true });
-    const mosaicPath = path.join(mosaicDir, dummyImageName);
-    const dummyImagePath = '/home/worker/dummy.tif';
-    log('Copy dummy image to datastore directory');
-    await fsPromises.copyFile(dummyImagePath, mosaicPath);
 
-    log('Create image mosiaic store via REST');
-    await grc.datastores.createImageMosaicStore(ws, covStore, zipPath);
+    log('Copy prototype granule to datastore directory');
+    const prototypeGranuleName = path.basename(prototypeGranule);
+    const mosaicPath = path.join(mosaicDir, prototypeGranuleName);
+    await fsPromises.copyFile(prototypeGranule, mosaicPath);
 
-    log('delete dummy granule from database and from disc');
-    await grc.imagemosaics.deleteSingleGranule(ws, covStore, covStore, dummyImageName);
-    await fsPromises.rm(mosaicPath);
+    log('Create image mosaic store via REST');
+    await grc.datastores.createImageMosaicStore(ws, covStore, zipPath, false);
 
     log(`... CoverageStore ${covStore} created`);
 
+    log('Initialize the store');
+    // Once a prototype has been provided we need to initialize the store by querying it for the available coverages.
+    // TODO check if really is neccessary
+    // http://localhost:8080/geoserver/rest/workspaces/{ws}/coveragestores/{covname}/coverages.json?list=all
+
+    log(`Enabling time for layer "${ws}:${covStore}"`);
+    await grc.layers.enableTimeCoverage(ws, covStore, covStore, 'LIST', 3600000, 'MAXIMUM', true, false, 'PT30M');
```

Comment:
time periodicity on 1 hour is more realistic I think

```suggestion
    await grc.layers.enableTimeCoverage(ws, covStore, covStore, 'LIST', 3600000, 'MAXIMUM', true, false, 'PT1H');
```

Reply:
Ok, but since the pre-last parameter is false (`rawNearestMatchEnabled`), this last param is obsolete anyway. Will leave this for now, if we use `nearest match` should be discussed later.